src/controller/java/CHIPDeviceController-JNI.cpp

```diff
@@ -699,6 +705,12 @@ JNI_METHOD(void, pairDeviceWithAddress)
         JniByteArray jniCsrNonce(env, csrNonce);
         commissioningParams.SetCSRNonce(jniCsrNonce.byteSpan());
     }
+
+    if (icdRegistrationInfo != nullptr)
+    {
+        wrapper->ApplyICDRegistrationInfo(commissioningParams, icdRegistrationInfo);
+    }
```

Comment:
I'm not sure i understand why this present in multiple locations?

Reply:
There are three commissioning APIs. Therefore, it was necessary to parse ICD Registration Information for each API, so it was added.